pennylane/pulse/transmon.py

```diff
@@ -215,3 +222,264 @@ def __add__(self, other):
+
+    The Hamiltonian is given by
+
+    .. math::
+
+        \Omega(t) \left(e^{i (\phi(t) + \nu t)} a_q + e^{-i (\phi(t) + \nu t)} a^\dagger_q\right)
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators
+    and :math:`q` is the qubit label (``wires``).
+    The arguments ``amplitude``, ``phase`` and ``freq`` correspond to :math:`\Omega`, :math:`\phi`
+    and :math:`\nu`, respectively, and can all be either fixed numbers (``float``) or depend on time
+    (``callable``). If they are time-dependent, they need to abide by the restrictions imposed
+    in :class:`ParametrizedHamiltonian` and have a signature of two parameters, ``(params, t)``.
+
+    For realistic simulations, one may restrict the amplitude, phase and drive frequency parameters.
+    For example, the authors in `2008.04302 <https://arxiv.org/abs/2008.04302>`_ impose the restrictions of
+    a maximum amplitude :math:`\Omega_{\text{max}} = 20 \text{MHz}` and the carrier frequency to deviate at most
+    :math:`\nu - \omega = \pm 1 \text{GHz}` from the qubit frequency :math:`\omega`
+    (see :func:`~.transmon_interaction`).
+    The phase :math:`\phi(t)` is typically a slowly changing function of time compared to :math:`\Omega(t)`.
+
+    .. note:: Currently only supports ``d=2`` with qudit support planned in the future. For ``d=2`` we have :math:`a:=\frac{1}{2}(\sigma^x + i \sigma^y)`.
+
+    .. note:: Due to convention in the respective fields, we ommit the factor :math:`\frac{1}{2}` present in the related constructors :func:`~.drive` and :func:`~.rydberg_drive`
+
+    .. seealso::
+
+        :func:`~.drive`, :func:`~.rydberg_drive`, :func:`~.transmon_interaction`
+
+    Args:
+        amplitude (Union[float, callable]): The amplitude :math:`\Omega(t)` (in GHz).
+            Can be a fixed number (``float``) or depend on time (``callable``)
+        phase (Union[float, callable]): The phase :math:`\phi(t)` (in radians).
+            Can be a fixed number (``float``) or depend on time (``callable``)
+        freq (Union[float, callable]): The drive frequency :math:`\nu` (in GHz).
+            Can be a fixed number (``float``) or depend on time (``callable``)
+        wires (Union[int, list[int]]): Label of the qubit that the drive acts upon. Can be a list of multiple wires.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        :class:`~.ParametrizedHamiltonian`: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    We can construct a drive term acting on qubit ``0`` in the following way. We parametrize the amplitude and phase
+    via :math:`\Omega(t) = \Omega \sin(\pi t)` and :math:`\phi(t) = \phi (t - \frac{1}{2})`.
+
+    .. code-block:: python3
+
+        def amp(Omega, t): return Omega * jnp.sin(jnp.pi*t)
```

Comment:
```suggestion
        def amp(max_amp, t): return max_amp * jnp.sin(jnp.pi*t)
```

Or similar - I think it's confusing that there is a variable called `omega` and another called `Omega` and they are different.

Reply:
I like it ✅